src/tracks/PatternTrack.cpp

```diff
@@ -149,14 +149,15 @@ void PatternTrack::saveTrackSpecificSettings(QDomDocument& doc, QDomElement& _th
 /*	_this.setAttribute( "current", s_infoMap[this] ==
 					engine::getPatternEditor()->currentPattern() );*/
 	if( s_infoMap[this] == 0 &&
-			_this.parentNode().parentNode().nodeName() != "clone" &&
+			_this.parentNode().parentNode().nodeName() != "clonedtrack" &&
 			_this.parentNode().parentNode().nodeName() != "journaldata" )
 	{
 		Engine::patternStore()->saveState(doc, _this);
 	}
-	if( _this.parentNode().parentNode().nodeName() == "clone" )
+	// If we are creating drag-n-drop data for Track::clone() only save pattern ID, not pattern content
+	if (_this.parentNode().parentNode().nodeName() == "clonedtrack")
 	{
-		_this.setAttribute( "clonebbt", s_infoMap[this] );  // TODO rename bb to pattern
```

Comment:
yesSSss well... `s_infoMap[this]` is the same as `patternIndex()` which is what the PatternTrack uses to select the right thing from PatternStore. So both options are correct, but I'd still let's stick with `sourcepattern`.

We should probably just replace all `s_infoMap[this]` with `patternIndex()` because the former one is extremely obscure.

Reply:
Agreed to both (and make `patternIndex()` const?).

Reply:
In this PR?

Reply:
Why not? It's a non-functional change, right?